### ISSUE TITLE
Html text disappears from pdf

### ISSUE THREAD

**Issue by GeyseR** [Kozea/WeasyPrint#320]
Hi there!

I hope someone can help with weird weasyprint behaviour, what i have found.
I want to convert next html document to pdf:
```html
<!DOCTYPE html>
<html>
  <head>
    <style>
      @page {
        size: A4 portrait;
        margin: 0;
        border: 8.5mm #222b67 solid;
      }
      .working {
        border: 1px solid green;
      }
      .broken {
        border: 1px solid red;
      }
    </style>
  </head>
  <body>
    <div class="working">
      Some example text Some example text Some example text Some example tex
    </div>
    <div class="broken">
      Some example text Some example text Some example text Some example text Some example text
    </div>
    <div class="broken">
      Some example text Some example text Some example text Some example text Some
    </div>
  </body>
</html>
```
But inside result pdf text in broken divs just disappeared.
<img width="824" alt="broken-pdf" src="https://cloud.githubusercontent.com/assets/1364479/15785113/4ca90d3c-29be-11e6-9de1-b43754200488.png">

What i'm doing wrong? Can it be related to some internal bug in weasyprint or in dependencies?

**Comment by WnP** [comment]
same issue here, rendering to png works as expected but some text disapear in pdf rendering

@GeyseR did you find a fix?

**Comment by GeyseR** [comment]
Hi, @WnP!

Yeah, our team found workaround that helped us fix bud with disappeared text. We added to our html next css fragment:
```css
body {
  word-spacing: 0.1em;
}
```

Looks like dirty hack, still need to be fixed in WeasyPrint...

**Comment by liZe** [comment]
> Can it be related to some internal bug in weasyprint or in dependencies?

Yes, if the PNG is OK but the PDF is broken, it's probably a bug in Cairo. I can't reproduce the bug myself on Linux with Cairo 1.14.6.

If you're using an older version of Cairo, you should update it and see if it fixes the bug. If it doesn't then it's probably related to the OS you're using, and we should report the bug to the Cairo team.

**Comment by WnP** [comment]
@GeyseR thanks a lot, it woks like a charm

@liZe I've tested it using cairo 1.14.0, which is actualy the debian stable package `libcairo2 version: 1.14.0-2.1+deb8u1`, thanks for pointing the working version

**Comment by liZe** [comment]
This bug is actually a duplicate of #233. It's been fixed in Cairo 1.14.2.
